SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -349,6 +535,66 @@ void HGCDigitizer::accumulate(PileUpEventPrincipal const& e,
   }
 }
 
+//
+template <typename GEOM>
+void HGCDigitizer::accumulate_forPreMix(edm::Handle<edm::PCaloHitContainer> const& hits,
+                                        int bxCrossing,
+                                        const GEOM* geom,
+                                        CLHEP::HepRandomEngine* hre) {
+  if (nullptr == geom)
+    return;
+
+  float keV2fC(0.f);
+
+  int nchits = (int)hits->size();
+
+  std::vector<HGCCaloHitTuple_t> hitRefs;
+  hitRefs.reserve(nchits);
+  for (int i = 0; i < nchits; ++i) {
+    const auto& the_hit = hits->at(i);
+    DetId id = simToReco(geom, the_hit.id());
+    // to be written the verbosity block
+    if (id.rawId() != 0) {
+      hitRefs.emplace_back(i, id.rawId(), (float)the_hit.time());
+    }
+  }
+  std::sort(hitRefs.begin(), hitRefs.end(), this->orderByDetIdThenTime);
+
+  nchits = hitRefs.size();
+  for (int i = 0; i < nchits; ++i) {
+    const int hitidx = std::get<0>(hitRefs[i]);
+    const uint32_t id = std::get<1>(hitRefs[i]);
+    if (!validIds_.count(id))
+      continue;
+
+    HGCPUSimHitDataAccumulator::iterator simHitIt = pusimHitAccumulator_->emplace(id, HGCCellHitInfo()).first;
+    if (id == 0)
+      continue;
+
+    const float toa = std::get<2>(hitRefs[i]);
+    const PCaloHit& hit = hits->at(hitidx);
+    const float charge = hit.energy() * 1e6 * keV2fC;  // * getCCE(geom, id, cce_);
+
+    const float dist2center(getPositionDistance(geom, id));
+    const float tof = toa - dist2center / refSpeed_ + tofDelay_;
+    const int itime = std::floor(tof / bxTime_) + 9;
+
+    if (itime < 0 || itime > 14)
```

Comment:
@kpedro88 , O don't understand this comment.

Reply:
14 is a "magic number": its meaning may not be clear to any reader/reviewer, and it also isn't clear if other instances of the number 14 in this file are the same as this one. For example, if I wanted to change this to 15, should I make any other changes elsewhere in the code?

It is always preferable to use a named constant, so that meaning and consistency are automatically enforced in the source code. This is CMS code rule 7.6: http://cms-sw.github.io/cms_coding_rules.html

Reply:
So, 14 is not actually a magic number per se. It's just that the HGCDigitizer is designed to capture hits from total 15 Bxs. And directly using the number 14 was the norm in the https://github.com/cms-sw/cmssw/blob/master/SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc#L421 and I just copied it. Probably would have to consult the DPG group before naming this as a constant magic number.

Reply:
Yes, HGCDigitizer needs a lot of cleanup. It would be great to improve the situation in this PR, but if not it can be addressed in a subsequent PR.